### ISSUE TITLE
webapp generator requires <finalName> to be set

### ISSUE THREAD

**Issue by rhuss** [fabric8io/fabric8-maven-plugin#1276]
### Description

The build must contain a 

```xml
 <build>
    <finalName>${project.artifactId}</finalName>
    ....
```

because otherwise the "webapp" assembly used in the generator doesn't find the war file to add. The generator should be smarter and really consider ${project.build.finalName} for adding the proper war-file

### Info

* f-m-p version : 3.5.38
* Maven version (`mvn -v`) :

```
Apache Maven 3.5.2 (138edd61fd100ec658bfa2d307c43b76940a5d7d; 2017-10-18T09:58:13+02:00)
Maven home: /Users/roland/.sdkman/candidates/maven/current
Java version: 1.8.0_171, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_171.jdk/Contents/Home/jre
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "10.13.2", arch: "x86_64", family: "mac"
```

**Comment by rohanKanojia** [comment]
@rhuss : I've discussed one temporary solution with you(not sure it's right approach though) on IRC, If `<finalName>` is not provided, maven auto-generates it and we can check for `<finalName>` by checking whether it's in an auto generated phase or not. Although it's more of a hack and I also don't feel it's clean solution.

Is there some other way via which we can validate `<finalName>`? WDYT?

**Comment by rohanKanojia** [comment]
@rhuss: Actually I wanted little help from your side so that I could have figured it out myself(that's why I assigned issue to myself). Anyways thanks for PR :)

**Comment by rhuss** [comment]
well the simple fix was already mentioned more or less in the problem description: Change the assembly webapp.xml to use `${project.build.finalName}` as the file to include.

However, an even better solution is to use a dependencySet instead of a fileSet and while I was at it, I found out that the Route generated for the Service for webapp doesn't really work nicely as it includes the artefact name + version as context path in the URL. So I added the `path` element to explicitly set the path which accidentally fixes this issue here, too. It was not my intention to take this issue away from you.
